caikit/runtime/client/remote_module_base.py

```diff
@@ -0,0 +1,496 @@
+
+    ### HTTP Helper Functions
+    def _request_via_http(
+        self,
+        method: RemoteRPCDescriptor,
+        service_type: ServiceType,
+        *args,
+        **kwargs,
+    ) -> Any:
+        # Get request data model
+        request_dm = DataBase.get_class_for_name(method.request_dm_name)(
+            *args, **kwargs
+        )
+
+        # ! This is a hack to ensure all fields/types have been json encoded (bytes/datetime/etc).
+        request_dm_dict = json.loads(request_dm.to_json(use_oneof=True))
+
+        # Parse generic Request type into HttpRequest format
+        if service_type == ServiceType.INFERENCE:
+            http_request_dict = {
+                REQUIRED_INPUTS_KEY: {},
+                OPTIONAL_INPUTS_KEY: {},
+                MODEL_ID: self._model_name,
+            }
+            for param in method.signature.parameters:
+                value = request_dm_dict.get(param)
+
+                # If param doesn't have a default then add it to inputs
+                if param not in method.signature.default_parameters:
+                    http_request_dict[REQUIRED_INPUTS_KEY][param] = value
+
+                # If the param is different then the default then add it to parameters
+                elif value != method.signature.default_parameters.get(param):
+                    http_request_dict[OPTIONAL_INPUTS_KEY][param] = value
+
+            # If there is only one input then collapse down the value
+            if len(http_request_dict[REQUIRED_INPUTS_KEY]) == 1:
+                http_request_dict[REQUIRED_INPUTS_KEY] = list(
+                    http_request_dict[REQUIRED_INPUTS_KEY].values()
+                )[0]
+        elif service_type == ServiceType.TRAINING:
+            # Strip all null values
+            def _remove_null_values(_attr):
+                if isinstance(_attr, dict):
+                    return {
+                        key: _remove_null_values(value)
+                        for key, value in _attr.items()
+                        if value
+                    }
+                if isinstance(_attr, list):
+                    return [
+                        _remove_null_values(listitem) for listitem in _attr if listitem
+                    ]
+
+                return _attr
+
+            http_request_dict = _remove_null_values(request_dm_dict)
+
+        # Get request options and target
+        request_kwargs = {"headers": {"Content-type": "application/json"}}
+        if self._tls.enabled:
+            if self._tls.cert_file and self._tls.key_file:
+                request_kwargs["cert"] = (
+                    self._tls.cert_file,
+                    self._tls.key_file,
+                )
+
+            if self._tls.insecure_verify:
+                request_kwargs["verify"] = False
+            else:
+                request_kwargs["verify"] = self._tls.ca_file or True
+
+        if method.output_streaming:
```

Comment:
I know we don't currently support `input_streaming` for HTTP, but we should probably at least leave a TODO here to support it if/when the runtime http server does